registrations/serializers.py

```diff
@@ -274,14 +273,131 @@ def validate(self, data):
 
 class SeatReservationCodeSerializer(serializers.ModelSerializer):
     timestamp = DateTimeField(default_timezone=pytz.UTC, required=False)
+
     expiration = serializers.SerializerMethodField()
 
-    class Meta:
-        fields = ("seats", "code", "timestamp", "registration", "expiration")
-        model = SeatReservationCode
+    in_waitlist = serializers.SerializerMethodField()
 
     def get_expiration(self, obj):
-        return obj.timestamp + timedelta(minutes=code_validity_duration(obj.seats))
+        return obj.expiration
+
+    def get_in_waitlist(self, obj):
+        registration = obj.registration
+        maximum_attendee_capacity = registration.maximum_attendee_capacity
+
+        if maximum_attendee_capacity is not None:
+            attendee_count = registration.signups.filter(
+                attendee_status=SignUp.AttendeeStatus.ATTENDING
+            ).count()
+
+            if maximum_attendee_capacity - attendee_count <= 0:
+                return True
+
+        return False
+
+    def validate(self, data):
+        instance = self.instance
+        errors = {}
+
+        # Amount of seats is always required
+        if data.get("seats") is None:
+            errors["seats"] = ErrorDetail(
+                _("This field must be specified."), code="required"
+            )
+
+        if isinstance(instance, SeatReservationCode):
+            non_editable_fields = ["registration", "timestamp"]
+
+            # It's not allowed to change registration or timestamp values
+            for field in non_editable_fields:
+                if field in data and getattr(instance, field) != data[field]:
+                    errors[field] = _("You may not change the value.")
+
+            # The code must be defined and match instance code when updating existing seats reservation
+            if self.initial_data.get("code") is None:
+                errors["code"] = ErrorDetail(
+                    _("This field must be specified."), code="required"
+                )
+            elif str(instance.code) != self.initial_data["code"]:
+                errors["code"] = _("The value doesn't match.")
+
+        registration = data["registration"]
+        maximum_attendee_capacity = registration.maximum_attendee_capacity
+        waiting_list_capacity = registration.waiting_list_capacity
+
+        # Validate attendee capacity only if maximum_attendee_capacity is defined
+        if maximum_attendee_capacity is not None:
+            serialized_registration = RegistrationBaseSerializer(
+                data["registration"], context=self.context
+            ).data
+            attendee_count = serialized_registration["current_attendee_count"]
```

Comment:
Very good point. Added `current_attendee_count`, `current_waiting_list_count`, `remaining_attendee_capacity` and `remaining_waiting_list_capacity` properties to Registration model instead of having similar methods in RegistrationSerializer